test/forms/behaviours/StringFieldBehaviours.scala

```diff
@@ -18,35 +18,55 @@ package forms.behaviours
 
 import org.scalacheck.Gen
 import play.api.data.{Field, Form, FormError}
+import wolfendale.scalacheck.regexp.RegexpGen
 
 trait StringFieldBehaviours extends FieldBehaviours {
 
-  def fieldWithMaxLength(form: Form[?], fieldName: String, maxLength: Int, lengthError: FormError): Unit =
+  def fieldWithMaxLength(form: Form[?], fieldName: String, maxLength: Int, lengthError: FormError, regex: String): Unit =
     s"must not bind strings longer than $maxLength characters" in {
 
-      forAll(stringsLongerThan(maxLength)) {
+      forAll(
+        stringsLongerThan(maxLength,
+                          RegexpGen
+                            .from(regex)
+                            .suchThat(_.nonEmpty)
+                            .flatMap(
+                              s => Gen.oneOf(s.toList)
+                            )
```

Comment:
Rather than this would it be easier to just change the default value for `charGen` from `arbitrary[Char]` to `Gen.alphaNumChar?`